google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java

```diff
@@ -278,28 +300,46 @@ public void onFacetInitialized(@NotNull Facet facet) {
 
     private final Project project;
     private final String fileName;
-    private final TextFieldWithBrowseButton filePicker;
+    private final TextFieldWithBrowseButton directoryPicker;
     private final Consumer<Path> configFileGenerator;
+    private final boolean isDirectory;
     // Used to refresh the warnings.
     private final Runnable configurationValidator;
 
     GenerateConfigActionListener(
         Project project,
         String fileName,
         Consumer<Path> configFileGenerator,
-        TextFieldWithBrowseButton filePicker,
+        TextFieldWithBrowseButton directoryPicker,
+        boolean isDirectory,
```

Comment:
ok. Ideally, this would be refactored a somehow, but if it ends up not being cleaner, then at a minimum I'd suggest to document the behavior of this class then since it takes a little effort to decipher as is - what the outcome of setting `isDirectory` one way or the other etc.